autoload/go/test_test.vim

```diff
@@ -0,0 +1,109 @@
+        \ {'lnum': 15, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'badness'},
+        \ {'lnum': 16, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'helper badness'},
+        \ {'lnum': 20, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'this is an error'},
+        \ {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'and a second line, too'},
+        \ {'lnum': 25, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'this is a sub-test error'},
+        \ {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'and a second line, too'},
+        \ {'lnum': 40, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'panic: concurrent fail'}
+      \ ]
+
+  call s:test(expected, 0)
+endfunc
+
+func! Test_GoTestVerbose() abort
+  let expected = [
+        \ {'lnum': 12, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'log message'},
+        \ {'lnum': 13, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'sub badness'},
+        \ {'lnum': 15, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'badness'},
+        \ {'lnum': 16, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'helper badness'},
+        \ {'lnum': 20, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'this is an error'},
+        \ {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'and a second line, too'},
+        \ {'lnum': 25, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'this is a sub-test error'},
+        \ {'lnum': 0, 'bufnr': 0, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'and a second line, too'},
+        \ {'lnum': 31, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'goodness'},
+        \ {'lnum': 40, 'bufnr': 2, 'col': 0, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': 'panic: concurrent fail'}
+      \ ]
+
+  call s:test(expected, 1)
+endfunc
+
+func! s:test(expected, verbose) abort
+  if has('nvim')
+    " don't test neovim; it seems to mostly work, but the errors it displays
+    " are slightly out of order due to the way it interleaves stderr and
+    " stdout lines from async jobs, and it doesn't implement support for
+    " 'errorformat' correctly; multi-line errors get handled differently than
+    " they do for Vim, and the messages produce by neovim include characters
+    " that get trimmed out of Vim's messages.
```

Comment:
Does this mean that `:GoTest` on NeoVim is broken, or just that it's hard to write tests for NeoVim?

Reply:
Neither, actually. Neovim's implementation of errorformat works, but results in slightly different results. Honestly, I'm ready to drop support for Neovim.

Reply:
> Honestly, I'm ready to drop support for Neovim.

Aren't a lot of people using NeoVim?

Reply:
I'm sure they are, but first-class support in vim-go for Neovim is explicitly *not* a goal for vim-go, and there are a lot of special cases in the code for it. Neovim isn't Vim, and it just causes clutter to pretend that it is. 

YMMV, and I'll continue to support it as best we can until we decide to do otherwise.

Reply:
Neovim adoption is growing and many people using Neovim are using vim-go as well. If it works in Neovim, we shouldn't break it, especially for main commands such as `:GoTest`. I know it's hard to maintain both Vim and Neovim, part of me wants to get rid of Neovim as it's hard to keep up with two editors, but realities are different

Reply:
Yeah, I get it. I've tried hard to make sure this PR won't break NeoVim handling of `:GoTest` output; it significantly better, but it's a little different than Vim's.